### PR TITLE
Basic functionality code

### PR DESCRIPTION
**Pull request by haripallikere**
* Contains all the code that was used for the creation of image slider

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  //populates images to dom
+  generateImages() {
+    this.slide.innerHTML = this.images
+      .map((v, i) => {
+        if (i <= 1) {
+          return `<img src="${v}" id="${i}">`;
+        }
+        if (i >= 2) {
```
Comment (rahulloconav) [inline main.js]: A simple else block would be preferred.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      .join("");
+  }
+
+  //populate indicators
+  generateIndicators() {
+    const removeClonedImages = [...this.images];
+    removeClonedImages.shift();
+    removeClonedImages.pop();
+    this.indicatorsContainer.innerHTML = removeClonedImages
+      .map((v, index) => {
```
Comment (rahulloconav) [inline main.js]: .map((v, index) => `<button class="navigation__indicators" id="${index}"></button>`)

Above code looks more readable

File: style.css [hunk 3]
```diff
@@ -93,3 +93,119 @@ main {
...
+
+.slider__btn {
+  border: 10px solid red;
+  border: none;
+  height: 60px;
+  width: 60px;
+  text-align: center;
+  position: absolute;
+  z-index: 1000;
+  top: 48%;
```
Comment (rahulloconav) [inline style.css]: Instead of top: 48% try:
`top: 50%;
transform: translateY(-50%)`
This would make it perfectly centered.

File: style.css [hunk 4]
```diff
@@ -93,3 +93,119 @@ main {
...
+.navigation__bar {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 50px;
+}
+
+.navigation__indicators {
+  margin-left: 20px;
```
Comment (rahulloconav) [inline style.css]: instead of using margin-left to seperate flex-items use `row-gap or column gap property` in `.navigation__bar`

**Review by rahulloconav — CHANGES_REQUESTED** [review]

File: main.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+      })
+      .join("");
+  }
+
+  //populate indicators
+  generateIndicators() {
+    const removeClonedImages = [...this.images];
+    removeClonedImages.shift();
+    removeClonedImages.pop();
+    this.indicatorsContainer.innerHTML = removeClonedImages
```
Comment (abidRahim) [inline main.js]: Let's try to use an alternative to innerHTML because it has security flaws

File: main.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    this.images = images;
+    this.document = document;
+    this.isTransition = false;
+    this.indicatorsContainer = indicatorsContainer;
+    this.slideIndex = 1;
+  }
+
+  //populates images to dom
+  generateImages() {
+    this.images.map((v, i) => {
```
Comment (abidRahim) [inline main.js:31]: `this.images.map((img, ind) => {`
will be clear in understanding the code. Avoid single letter variables

File: style.css [hunk 7]
```diff
@@ -93,3 +93,120 @@ main {
...
+  line-height: 1.2rem;
+  color: wheat;
+}
+
+.slider {
+  display: block;
+  /* margin: auto; */
+  position: relative;
+  /* display: grid; */
+  /* place-items: center; */
```
Comment (abidRahim) [inline style.css:114]: remove commented css

File: style.css [hunk 8]
```diff
@@ -93,3 +93,120 @@ main {
...
+
+.slider:hover .slider__btn {
+  display: block;
+}
+
+
+
+
+
+
```
Comment (abidRahim) [inline style.css:209]: remove extra spaces

File: main.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+  // move images
+  moveSlide() {
+    this.lazyLoading();
+    this.slide.style.transform = `translateX(-${this.slideIndex * 100}%)`;
+  }
+
+  // onclick
+  moveHandler(direction) {
+    this.isTransition = true;
+    this.slide.style.transition = "transform .6s ease-in-out";
```
Comment (hemkaran) [inline main.js:67]: Try to use single quotes in string in javascript.
Use double quotes in html and jsx

File: main.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        img.setAttribute("class", "lazy");
+        img.setAttribute("data-src", `${v}`);
+      }
+      this.slide.append(img);
+    });
+  }
+
+  //populate indicators
+  generateIndicators() {
+    const removeClonedImages = [...this.images];
```
Comment (hemkaran) [inline main.js:47]: Why the variable name is `remove`?

File: main.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+  moveSlide() {
+    this.lazyLoading();
+    this.slide.style.transform = `translateX(-${this.slideIndex * 100}%)`;
+  }
+
+  // onclick
+  moveHandler(direction) {
+    this.isTransition = true;
+    this.slide.style.transition = "transform .6s ease-in-out";
+    if (direction === "right") {
```
Comment (hemkaran) [inline main.js:68]: Try to use `enum`  at such places.
```
const DIRECTION = { RIGHT: 'right', LEFT: 'left' }
```

File: main.js [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+      this.slideIndex += 1;
+      this.moveSlide();
```
Comment (hemkaran) [inline main.js:70]: `moveSlide` can be moved outside.

File: main.js (hunk 6 above)
Comment (haripallikere) [reply main.js:31]: resolved

File: style.css (hunk 7 above)
Comment (haripallikere) [reply style.css:114]: resolved

File: main.js (hunk 10 above)
Comment (haripallikere) [reply main.js:47]: rectified

File: main.js (hunk 9 above)
Comment (haripallikere) [reply main.js:67]: resolved

**Pull request merged**